File: Python/reverse_linked_list.py
""" Reverse a singly linked list """
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reverseList(self, head):
        current = head
        temp = current

        if head is None:
            return None
        if head.next is None:
            return head
        while (head != None):
           temp = head
           head = head.next
           temp.next = current
           current = temp
        return current

File: C/reverse_linked_list.c
// Reverse a singly linked list.

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
  struct ListNode* current = NULL;
  struct ListNode* temp = NULL;

  if (!head) {
    return NULL;
  } 
  while(head) {
    temp = head;
    head = head->next;
    temp->next = current;
    current = temp;
  }
  return current;
}
